src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -845,30 +820,16 @@ internal JsonPropertyDictionary<JsonPropertyInfo> CreatePropertyCache(int capaci
         // - class type,
         // - element type (if the type is a collection),
         // - the converter (either native or custom), if one exists.
-        internal static JsonConverter GetConverter(
-            Type type,
-            Type? parentClassType,
-            MemberInfo? memberInfo,
+        private protected static JsonConverter GetConverterFromMember(
+            Type typeToConvert,
+            MemberInfo memberInfo,
             JsonSerializerOptions options,
             out JsonConverter? customConverter)
         {
-            Debug.Assert(type != null);
-            Debug.Assert(!IsInvalidForSerialization(type), $"Type `{type.FullName}` should already be validated.");
-            customConverter = parentClassType != null ? options.GetCustomConverterFromMember(parentClassType, type, memberInfo) : null;
-            return options.GetConverterForType(type);
-        }
-
-        internal static JsonConverter GetEffectiveConverter(
```

Comment:
Method removed since it was always invoked with `memberInfo == null`. Replaced with direct invocation to `options.GetConverterForType`.